oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/DetailPageEditForm.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useRef } from "react";
+import PropTypes from "prop-types";
+import { Container, Grid, Sticky, Ref } from "semantic-ui-react";
+import { BaseForm, TextField } from "react-invenio-forms";
+import { PublishButton } from "./components/PublishButton";
+import { MultiLingualTextInput } from "./components/MultiLingualTextInput";
+import { PropFieldsComponent } from "./components/PropFieldsComponent";
+import { useLocation } from "react-router-dom";
+import { ResetButton } from "./components/ResetButton";
+import { VocabularyFormSchema } from "./VocabularyFormSchema";
+import { CurrentLocationInformation } from "./components/CurrentLocationInformation";
+import { useFormConfig } from "@js/oarepo_ui/forms";
+import _omitBy from "lodash/omitBy";
+import Overridable from "react-overridable";
+import { ApiClientInitialized } from "@js/oarepo_ui/api";
+import { ErrorElement } from "@js/oarepo_ui/search";
+import { useMutation } from "@tanstack/react-query";
+
+const removeNullAndUnderscoreProperties = (obj) => {
+  return _omitBy(
+    obj,
+    (value, key) =>
+      value === null ||
+      (Array.isArray(value) && value.every((item) => item === null)) ||
+      key.startsWith("_")
+  );
+};
+
+export const DetailPageEditForm = ({
+  initialValues,
+  options,
+  hasPropFields,
+  editMode,
+  apiCallUrl,
+}) => {
+  const {
+    formConfig: { vocabularyProps },
+  } = useFormConfig();
+  const sidebarRef = useRef(null);
+  const location = useLocation();
+  const currentPath = location.pathname;
+  const searchParams = new URLSearchParams(location.search);
+  const newChildItemParentId = searchParams.get("h-parent");
+
+  const { error: saveError, mutateAsync: saveMutateAsync } = useMutation({
+    mutationFn: async ({ apiCallUrl, editedItem }) =>
+      ApiClientInitialized.saveDraft(apiCallUrl, editedItem),
+  });
+  const { error: createError, mutateAsync: createMutateAsync } = useMutation({
+    mutationFn: async ({ apiCallUrl, newItem }) =>
+      ApiClientInitialized.createDraft(apiCallUrl, newItem),
+  });
+
+  const onSubmit = (values, formik) => {
+    let preparedValues = values;
+    if (newChildItemParentId)
+      preparedValues.hierarchy = { parent: newChildItemParentId };
+
+    if (editMode) {
+      saveMutateAsync({
+        apiCallUrl,
+        editedItem: removeNullAndUnderscoreProperties(preparedValues),
+      })
+        .then(() => {
+          formik.setSubmitting(false);
+          window.location.href = currentPath.replace("/edit", "");
```

Comment:
What if edit route changes or has some query params? Better way to navigate is to use useFormConfig's record ui links to detail page